crates/core/src/archiver/parent.rs

```diff
@@ -264,10 +264,15 @@ impl Parent {
     ) -> RusticResult<ItemWithParent<O>> {
         let result = match item {
             TreeType::NewTree((path, node, tree)) => {
+                // TODO!: This one is weird, it's working as well with `unwrap_or_default`
+                // Check if this is expected behaviour
+                #[allow(clippy::unwrap_used)]
                 let parent_result = self
                     .is_parent(&node, &tree)
-                    .map(|node| node.subtree.unwrap());
+                    .map_parent_result(|node| node.subtree.unwrap());
```

Comment:
I also renamed it now, because it actually took me a while to see, that this is a customized `map`